### PR TITLE
Simple script for checking permissions.

### PR DESCRIPTION
**Pull request by therearesomewhocallmetim**
Writes the result in the same format as the C++ unit tests so that we could write its results to the same log and parse it with the Jenkins scripts.
I did think about using unittest, but it is impossible to pass parameters to such tests.

### CODE REVIEW

File: tools/python_tests/permissions_test.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            permissions
+        )
+    )
+
+    if not permissions.issubset(EXPECTED_PERMISSIONS):
+        logging.info("Expected:\n{0}".format(EXPECTED_PERMISSIONS))
+        logging.info("Actual: \n{0}".format(permissions))
+        logging.info("FAILED")
+        logging.info("Test took {} ms".format((time.time() - start_time) * 1000))
+        exit(2)
```
Comment (bykoianko) [inline tools/python_tests/permissions_test.py]: Может напишем, какие новые пермишены добавлены, если тест не прошел? А то искать разницу в двух больших списках не оч удобно.

File: tools/python_tests/permissions_test.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        shell=True
+    )
+
+    out, _ = process.communicate()
+
+    return filter(None, out.splitlines())
+
+
+def find_aapt(args):
+    with open("../../android/local.properties") as props:
```
Comment (ygorshenin) [inline tools/python_tests/permissions_test.py]: Please, make a relative path from the script, not from the directory the script was executed in.

File: tools/python_tests/permissions_test.py (hunk 1 above)
Comment (ygorshenin) [inline tools/python_tests/permissions_test.py]: +1
We need two lists: lists of expected permissions that are absent, and list of unexpected permissions that are present.

**Comment by therearesomewhocallmetim** [comment]
JTTP FORMAT

File: tools/python_tests/permissions_test.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            lambda s: s.startswith("permission:") or s.startswith("uses-permission:"),
+            permissions
+        )
+    )
+
+    if permissions != EXPECTED_PERMISSIONS:
+        logging.info("\nExpected:\n{0}".format(new_lines(EXPECTED_PERMISSIONS)))
+        logging.info("\nActual: \n{0}".format(new_lines(permissions)))
+        logging.info(
+            "\nThe following permissions have been _added_:\n{0}".format(
```
Comment (ygorshenin) [inline tools/python_tests/permissions_test.py]: Why do you need underscores? Remove them, please.

Comment (therearesomewhocallmetim) [inline tools/python_tests/permissions_test.py]: I added the underscores to make the important words more visible.

File: tools/python_tests/permissions_test.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        dest="apk", default=None
+    )
+
+    return parser.parse_args()
+
+
+def exec_shell(executable, flags):
+    spell = ["{0} {1}".format(executable, flags)]
+    logging.info("Spell: {}".format(spell[0]))
+    process = subprocess.Popen(
```
Comment (Zverik) [inline tools/python_tests/permissions_test.py]: Я здесь не понял. Первым параметром же идёт массив из названия файла и каждого параметра отдельно, а не строка?

Comment (Zverik) [inline tools/python_tests/permissions_test.py]: А, у тебя shell=True. Понятненько.

**Comment by Zverik** [comment]
LGTM

File: tools/python_tests/permissions_test.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            )
+        )
+        logging.info(
+            "\nRemoved:\n{0}".format(
+                new_lines(EXPECTED_PERMISSIONS - permissions)
+            )
+        )
+        logging.info("FAILED")
+        logging.info("Test took {} ms".format((time.time() - start_time) * 1000))
+        exit(2)
```
Comment (ygorshenin) [inline tools/python_tests/permissions_test.py]: s/2/1/g

File: tools/python_tests/permissions_test.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    permissions = exec_shell(args.aapt, "dump permissions {0}".format(args.apk))
+    permissions = set(
+        filter(
+            lambda s: s.startswith("permission:") or s.startswith("uses-permission:"),
+            permissions
+        )
+    )
+
+    if permissions != EXPECTED_PERMISSIONS:
+        logging.info("\nExpected:\n{0}".format(new_lines(EXPECTED_PERMISSIONS)))
```
Comment (ygorshenin) [inline tools/python_tests/permissions_test.py]: Please, wrap this into a function that returns bool, and unify the code that measures the duration of the test.

**Comment by bykoianko** [comment]
LGTM

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**